Fix consistency and details in log in/create an account links

andreslucena opened the pull request:
#### :tophat: What? Why?

We have found some mini inconsistencies and changes to made in the 'log in' and 'create an account' pages in links:

- In the header it says "Login" - it should say "Log in"
- The icon for the Login is not the same in desktop and in mobile. The correct one is on mobile
- The "Create an account" page title is called "Sign up". It should say "Create an account".
- The links to that page should also say "Create an account" instead of "Sign up"

#### :pushpin: Related Issues
 
- Fixes #13060

#### Testing

1. Go to the login page
2. see the links
3. Go to the create an account page

### :camera: Screenshots

![Screenshot of the 'create an account' page](https://github.com/decidim/decidim/assets/717367/d4b8f5ff-6585-42f4-8a50-6c54b0859b76)

:hearts: Thank you!

alecslupu reviewed (changes requested):
The following key must be changed: `decidim.components.add_comment_form.account_message`, as it states: `<a href="%{sign_in_url}">Log in with your account</a> or <a href="%{sign_up_url}">sign up</a> to add your comment.`
![image](https://github.com/decidim/decidim/assets/105683/5b412fab-d7d9-4bf0-b9be-783d0b318f4b)

Also, in the footer we have "Sign up", i would expect to see "Create an account"
![image](https://github.com/decidim/decidim/assets/105683/9df27eb6-657e-40ac-8ec8-3a906ac91df3)

When the organization has force authentication enabled from system, the following message is displayed. I would expect to see "Please log in with your account before access"

![image](https://github.com/decidim/decidim/assets/105683/45a8941f-944d-4027-a9c5-02fbee29a34c)

I would expect "log in" in the below screen. 
![image](https://github.com/decidim/decidim/assets/105683/733175b0-dae8-4959-bbe1-bd76d9da15ad)

When trying to access an  area that requires authentication  ( like /admin), the following message is displayed. 
"You need to log in or sign up before continuing."
![image](https://github.com/decidim/decidim/assets/105683/12f0a197-16fb-4bce-9825-1dd8a6185533)

alecslupu reviewed (changes requested):
I would say there are still 2-3 places where the wording could be improved... 
Grepping revealed this: 

```
alecslupu@alecslupu-strix:~/Sites/decidim/redesign$ grep -RiHn 'sign up' ./decidim-core/config/locales/
./decidim-core/config/locales/en.yml:589:          subtitle: Please fill in the following form in order to complete the sign up
./decidim-core/config/locales/en.yml:1638:      unauthenticated: You need to log in or sign up before continuing.
./decidim-core/config/locales/en.yml:1861:      not_found: could not be found. Did you sign up previously?
```
I was not able to reveal the other 2 texts. Could you validate they are still being used, and provide a testing scenario? 

![image](https://github.com/decidim/decidim/assets/105683/b51fc322-a59b-44e9-8733-d79f0158999c)

andreslucena commented:
> ./decidim-core/config/locales/en.yml:589:          subtitle: Please fill in the following form in order to complete the sign up

Changed. Evidence in test: https://github.com/decidim/decidim/blob/d42f3af461959a9e00d2efe65b689988839ea500/decidim-core/spec/system/authentication_spec.rb#L147

> ./decidim-core/config/locales/en.yml:1638:      unauthenticated: You need to log in or sign up before continuing.

Changed. Evidence in test: https://github.com/decidim/decidim/blob/d42f3af461959a9e00d2efe65b689988839ea500/decidim-admin/spec/system/admin_access_spec.rb#L20

> ./decidim-core/config/locales/en.yml:1861:      not_found: could not be found. Did you sign up previously?

This was was tricky. This actually comes from devise/devise-i18n, and it gets kicked in with the unlock your account form when the email isn't found. The thing is that as we have the `config.paranoid = true` enabled by default, this isn't shown, as it would show possible user enumeration kind of attacks (i.e. I want to check out if an email is a registered account to try other kind of attacks).  Just in case, I changed it, as it exists the possibility of an implementer having this enabled. I think it's not needed a test for this case as it isn't the default configuration for Decidim, but if you feel it's necessary let me know and I can provide one. 

You can check this message out by changing the devise configuration and going to the unlock form with an email of a registered participant:

1. Apply this patch
```diff
diff --git a/decidim-core/config/initializers/devise.rb b/decidim-core/config/initializers/devise.rb
index 9dced963ff..c5c6a58344 100644
--- a/decidim-core/config/initializers/devise.rb
+++ b/decidim-core/config/initializers/devise.rb
@@ -89,7 +89,7 @@ Devise.setup do |config|
   # It will change confirmation, password recovery and other workflows
   # to behave the same regardless if the e-mail provided was right or wrong.
   # Does not affect registerable.
-  config.paranoid = true
+  config.paranoid = false

   # By default Devise will store the user in session. You can skip storage for
   # particular strategies by setting this option.
```
2. Go to http://localhost:3000/users/unlock
3. Enter "admin@example.org" and click the "Resend unlock instructions" button

alecslupu reviewed (approved):
:+1:
